subworkflows/local/fqtk_demultiplex/main.nf

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env nextflow
+
+//
+// Demultiplex Singular Genomics bases data using sgdemux
+//
+
+include { FQTK } from "../../../modules/nf-core/fqtk/main"
+
+workflow FQTK_DEMULTIPLEX {
+    take:
+        // Input fastq's must be bgzipped for compatibility with sgdemux
+        // samplesheet.csv must be a two column csv = Sample_Barcode,Sample_ID)
+        //TODO fix name custom_input
+        ch_input     // [[id:"", lane:""],samplesheet.csv, path/to/fastq/files]
+
+    main:
+        
+        // MODULE: fqtk
+        // Demultiplex the bases files
+        FQTK( ch_input )
+
+        // Generate meta for each fastq
+        ch_fastq_with_meta = generate_fastq_meta(FQTK.out.sample_fastq)
+
+    emit:
+        fastq                   = ch_fastq_with_meta
+       //generated_run_manifest  = FQTK.out.per_project_metrics 
+        metrics                 = FQTK.out.metrics 
+        unassigned              = FQTK.out.most_frequent_unmatched 
+        versions                = FQTK.out.versions
+}
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    FUNCTIONS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+// Add meta values to fastq channel
```

Comment:
I am not sure how to update these functions to best work with fqtk. It runs as is but will probably need some changes for all cases.